assets/js/googlesitekit/datastore/user/user-info.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * core/user Data store: user info
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+
+const { createRegistrySelector } = Data;
+
+const RECEIVE_USER_INFO = 'RECEIVE_USER_INFO';
+
+const INITIAL_STATE = {
+	user: undefined,
+	verified: undefined,
+};
+
+export const actions = {
+	/**
+	 * Stores user info in the datastore.
+	 *
+	 * Because this is frequently-accessed data, this is usually sourced
+	 * from a global variable (`_googlesitekitUserData`), set by PHP
+	 * in the `before_print` callback for `googlesitekit-datastore-user`.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} userInfo User info, usually supplied via a global variable from PHP.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveUserInfo( userInfo ) {
+		invariant( userInfo, 'userInfo is required.' );
+		const { user, verified } = userInfo;
+		return {
+			payload: {
+				user,
+				verified,
+			},
+			type: RECEIVE_USER_INFO,
+		};
+	},
+};
+
+export const controls = {};
+
+export const reducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case RECEIVE_USER_INFO: {
+			const { user, verified } = payload;
+			return {
+				...state,
+				user,
+				verified,
+			};
+		}
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+	*getUser() {
+		const { select } = yield Data.commonActions.getRegistry();
+
+		const { user, verified } = select( STORE_NAME ).getUser();
```

Comment:
Based on the other comments, this only should receive `user`, and shouldn't need to worry about `verified`.

Basically, both `getUser` and `isVerified` resolvers would result in the same logic, but since the check for whether a value is already set, that shouldn't be a problem. Plus it's not making an API request, so no real overhead anyway.